### PR TITLE
feat(button): added dropdown buttons

### LINKED ISSUE #112: [Feature Request] - Dropdown button
**Describe the functionality of the new component:**
A dropdown  button according to the Design system found in 
https://xd.adobe.com/view/a8299f86-4d90-40f9-4d23-7dc26f9266e3-de18/screen/eb6930f3-3b16-49ee-9aaa-3dbc780e7154/

**More details:**
Screenshot 2020-08-17 at 10 55 50 PM

### Properties
This will be part of fw-button itself
Property | Attribute | Description | Type | Default
---------|----------|------------|-------|--------
dropdown|dropdown|dropdown button or not|Boolean|false
split|split|Whether to split dropdown or not. Invalid if dropdown is false|Boolean|false
searchable|searchable|Whether the dropdown button should be searchable. Invalid if dropdown is false|Boolean|false
value|value|Selected options|Array|undefined
placeholder|placeholder|Placeholder for search input|String|''

### Events
Event | Description | Type
------|-------------|------
fwOptionClick|Emitted on selecting an option|customEvent
fwOptionsAdd|Emitted on adding options (`Add` button click)|customEvent

### Sample HTML
```html 
<fw-button dropdown split> Select OTT 
  <span id="1" value="netflix"> Netflix </span>
  <span id="2" value="amazon"> Amazon </span>
</fw-button>
```

### PR DESCRIPTION
**Pull request by shravan-balasubramanian**
Dropdown buttons with various settings - split, searchable

fix #112

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
Tested this with storybook and `npm run local`

### CODE REVIEW

File: src/components/button/button.scss [hunk 1]
```diff
@@ -22,133 +30,285 @@
...
+  }
+
+  100% {
+    opacity: 1;
+    transform: translateY(0%);
+  }
+}
+
+.btn-container {
+  display: flex;
```
Comment (asifahmedfw) [inline src/components/button/button.scss:125]: Caused a line break. Breaks normal functionality.

File: src/components/button/button.tsx [hunk 2]
```diff
@@ -59,30 +123,195 @@ export class Button {
...
+     */
+
+    if (this.isValidDropdown && !this.split) {
+      return this.handleDropdownToggle();
+    }
+
     this.fwClick.emit();
   }
 
+  private hasOptions() {
```
Comment (asifahmedfw) [inline src/components/button/button.tsx:152]: Can we use named slots and combine it with fw-option maybe? we can discuss on this.

**Review by asifahmedfw — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@freshworks/crayons",
   "engines": {
-    "node": "10.x"
+    "node": "10.18.1"
```
[the remark this answers is not in the document]
Comment (shravan-balasubramanian) [reply package.json:4]: I got an error telling `needs 10.181.`. Anyways will double-check on this

File: src/components/button/button.tsx (hunk 2 above)
Comment (shravan-balasubramanian) [reply src/components/button/button.tsx:152]: Is there an `fw-option`. Do you mean `fw-select-option`?

**Comment by shravan-balasubramanian** [comment]
Duplicated by #136

**Pull request closed without merging**